File: src/main/java/org/jukeboxmc/block/BlockStructureVoid.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemStructureVoid;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockStructureVoid extends Block {

    public BlockStructureVoid() {
        super( "minecraft:structure_void" );
    }

    @Override
    public ItemStructureVoid toItem() {
        return new ItemStructureVoid();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.STRUCTURE_VOID;
    }

    public void setStructureVoidType( StructureVoidType structureVoidType ) {
        this.setState( "structure_void_type", structureVoidType.toString() );
    }

    public StructureVoidType getStructureVoidType() {
        return this.stateExists( "structure_void_type" ) ? StructureVoidType.valueOf( this.getStringState( "structure_void_type" ) ) : StructureVoidType.VOID;
    }

    public enum StructureVoidType {
        VOID,
        AIR
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockStructureVoid.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.StructureVoid
import org.jukeboxmc.api.block.data.StructureVoidType
import org.jukeboxmc.server.block.JukeboxBlock

class BlockStructureVoid(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates),
    StructureVoid {

    override fun getStructureVoidType(): StructureVoidType {
        return StructureVoidType.valueOf(this.getStringState("structure_void_type"))
    }

    override fun setStructureVoidType(value: StructureVoidType): StructureVoid {
        return this.setState("structure_void_type", value.name.lowercase())
    }
}